src/Symfony/Component/DependencyInjection/Loader/XmlFileLoader.php

```diff
@@ -387,13 +387,7 @@ private function getArgumentsAsPhp(\DOMElement $node, $name, $lowercase = true)
                         $invalidBehavior = ContainerInterface::NULL_ON_INVALID_REFERENCE;
                     }
 
-                    if ($strict = $arg->getAttribute('strict')) {
```

Comment:
@stof I agree with you. But the same is true for the YAML format. So IMO it's inconsistent to deprecate this only in the YAML format. See #21058 to also trigger the deprecation with the XML format.